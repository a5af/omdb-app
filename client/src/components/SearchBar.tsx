import styles from './SearchBar.module.css';

import React, { useState, useRef, useEffect } from 'react';
import { useDispatch } from 'react-redux';

import { setResults } from '../features/search/searchSlice';
import { setConstantValue } from 'typescript';

const SearchBar = ({ API }: { API: string }) => {
  const [searchQ, setSearchQ] = useState('');
  const [results, setResults] = useState({});

  const [page, setPage] = useState(1);

  const ref = useRef<HTMLInputElement>();

  const dispatch = useDispatch();

  useEffect(() => {
    if (page && searchQ) {
      handleSubmit(false);
    }
  }, [page]);

  const handleSubmit = async (ev: any) => {
    console.log(searchQ, page);
    if (ev) {
      ev.preventDefault();
    }

    const response = await fetch(API, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        title: searchQ,
        page,
      }),
    });

    const responseJSON = await response.json();
    setResults(responseJSON);
  };

  const handleKey = async (ev: any) => {
    setSearchQ(ev.target.value);
  };

  const handlePageLast = () => {
    if (page > 1) setPage(page - 1);
  };

  const handlePageNext = () => {
    setPage(page + 1);
  };

  return (
    <div className={styles.SearchBar}>
      Search by Title
      <header className={styles.SearchBarHeader}>
        <input
          className={styles.SearchBarInput}
          onKeyUp={handleKey}
          placeholder={'Enter title'}
        ></input>
        <button
          disabled={searchQ == ''}
          className={styles.SearchBarSubmit}
          onClick={handleSubmit}
        >
          Search
        </button>

        <button
          disabled={searchQ == ''}
          className={styles.SearchBarSubmit}
          onClick={handlePageLast}
        >
          last
        </button>

        <button
          disabled={searchQ == ''}
          className={styles.SearchBarSubmit}
          onClick={handlePageNext}
        >
          next
        </button>
      </header>
      <div className={styles.Results}>{JSON.stringify(results)}</div>
    </div>
  );
};

export default SearchBar;
