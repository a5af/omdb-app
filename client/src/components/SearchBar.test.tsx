import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import SearchBar from './SearchBar';
import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import searchReducer from '../features/search/searchSlice';

export const store = configureStore({
  reducer: {
    searchReducer,
  },
});

it('changes the class when hovered', () => {
  const component = renderer.create(
    <Provider store={store}>
      <SearchBar API={'http://fake'} />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
