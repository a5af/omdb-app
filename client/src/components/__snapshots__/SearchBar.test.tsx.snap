// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="SearchBar"
>
  Search by Title
  <header
    className="SearchBarHeader"
  >
    <input
      className="SearchBarInput"
      onKeyUp={[Function]}
      placeholder="Enter title"
    />
    <button
      className="SearchBarSubmit"
      disabled={true}
      onClick={[Function]}
    >
      Search
    </button>
  </header>
  <div
    className="Results"
  >
    {}
  </div>
</div>
`;
